src/MainWindow.vala

```diff
@@ -1305,6 +1313,21 @@ namespace Terminal {
             }
         }
 
+
+        private void action_terminate () {
+            if (current_terminal.has_foreground_process ()) {
+                var d = new ForegroundProcessDialog.kill_process (this);
+
+                if (d.run () == Gtk.ResponseType.ACCEPT) {
```

Comment:
No need to repeat `d.destroy ()` - move after the if clause and remove else clause.